barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup_impl.hpp

```diff
@@ -749,6 +753,25 @@ element<C, Fq, Fr, G> element<C, Fq, Fr, G>::batch_mul(const std::vector<element
                                                        const bool with_edgecases)
 {
     auto [points, scalars] = handle_points_at_infinity(_points, _scalars);
+    OriginTag tag{};
+    const auto empty_tag = OriginTag();
+
+    // handle_points_at_infinity_method can remove some constant points, which messes with this code under
+    // CircuitSimulator
+    for (size_t i = 0; i < _points.size(); i++) {
+        tag = OriginTag(tag, OriginTag(_points[i].get_origin_tag(), _scalars[i].get_origin_tag()));
+    }
+    for (size_t i = 0; i < scalars.size(); i++) {
+        // If batch_mul actually performs batch multiplication on the points and scalars, subprocedures can do
+        // operations like addition or subtraction of points, which can trigger OriginTag security mechanisms even
+        // though the final result satisfies the security logic
+        // For example result = submitted_in_round_0 *challenge_from_round_0 +submitted_in_round_1 *
+        // challenge_in_round_1 will trigger it, because the addition of submitted_in_round_0 to submitted_in_round_1 is
```

Comment:
don't understand why this is dangerous

Reply:
We are going to detect when two submitted values from 2 rounds meet without a challenge in the future